### PR TITLE
replace examples in README with links to docs

### PR DESCRIPTION
**Pull request by rupa**
Some of the README examples are out-of-date/wrong. We didn't really
notice because the "real" docs are part of the repo and are published
on release, and we're not usuallty looking at the the README for a
source of truth.

Rather than cleaning up the README examples and having to keep them up
to date with future changes, just replace the examples with links to
the better maintained docs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,7 +30,8 @@ Building The Provider
 Clone repository to: `$GOPATH/src/github.com/terraform-providers/terraform-provider-ns1`
 
 ```sh
-$ mkdir -p $GOPATH/src/github.com/terraform-providers; cd $GOPATH/src/github.com/terraform-providers
+$ mkdir -p $GOPATH/src/github.com/terraform-providers
+$ cd $GOPATH/src/github.com/terraform-providers
```
Comment (mburtless) [inline README.md:34]: Is creating a home for the provider in $GOPATH necessary anymore, since we're using go mod?

Comment (rupa) [reply README.md:34]: i didn't wanna touch it cause i know we still ship a vendor dir around

**Comment by rupa** [comment]
yeah not sure why i didn't do that, we'll still forget to enumerate everything in the README. If I move the API links in each doc I think that'll do nicely

File: README.md (hunk 1 above)
Comment (mburtless) [reply README.md:34]: Ah gotcha, yeah I don't see any harm in leaving it as is.  Google's provider README still does it this way so 🤷‍♂️

**Pull request merged**